jobs/broken-site-report-ml/broken_site_report_ml/main.py

```diff
@@ -174,9 +225,64 @@ def get_missed_reports(client, last_run_time, bq_dataset_id):
     return list(query_job.result())
 
 
+def translate_by_uuid(client, uuids, bq_dataset_id):
+    uuids_sql = ", ".join(f"'{uuid}'" for uuid in uuids)
+
+    query = f"""
+            WITH reports AS (
+                SELECT uuid, comments as text_content
+                FROM
+                `moz-fx-data-shared-prod.org_mozilla_broken_site_report.user_reports_live`
+                WHERE uuid IN ({uuids_sql})
+            )
+            SELECT
+                uuid,
+                STRING(
+                    ml_translate_result.translations[0].detected_language_code
+                ) AS language_code,
+                STRING(
+                    ml_translate_result.translations[0].translated_text
+                ) AS translated_text,
+                ml_translate_status as status
+            FROM
+            ML.TRANSLATE(
+                MODEL `{bq_dataset_id}.translation`,
+                TABLE reports,
+                STRUCT(
+                  'translate_text' AS translate_mode,
+                  'en' AS target_language_code
+                )
+            );
+    """
+    query_job = client.query(query)
+    return list(query_job.result())
+
+
+def translate_reports(client, reports, bq_dataset_id):
+    result = {}
+    # Only translate reports that weren't translated
+    uuids_to_translate = [d["uuid"] for d in reports if not d["translated_text"]]
+
```

Comment:
If ML classification job fails due to bugbug not returning results in time, it will attempt to classify the same list of reports on the next run. In this case these reports will already be translated, so we don't need to translate them again and only query those without translation (`translated_text` is empty) by uuids.